apiml/src/main/java/org/zowe/apiml/ModulithConfig.java

```diff
@@ -0,0 +1,335 @@
+import org.springframework.cloud.netflix.eureka.server.event.EurekaRegistryAvailableEvent;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.event.EventListener;
+import org.springframework.http.server.reactive.HttpHandler;
+import org.springframework.http.server.reactive.TomcatHttpHandlerAdapter;
+import org.springframework.web.context.ServletContextAware;
+import org.zowe.apiml.discovery.ApimlInstanceRegistry;
+import org.zowe.apiml.filter.PreFluxFilter;
+import org.zowe.apiml.message.core.MessageService;
+import org.zowe.apiml.message.yaml.YamlMessageServiceInstance;
+import org.zowe.apiml.product.constants.CoreService;
+import reactor.core.publisher.Flux;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Configuration
+@RequiredArgsConstructor
+@EnableConfigurationProperties
+@Slf4j
+public class ModulithConfig {
+
+    private final ApplicationContext applicationContext;
+    private final Map<String, InstanceInfo> instances = new HashMap<>();
+    private final GatewayEurekaInstanceConfigBean eurekaInstanceGw;
+
+    @Value("${server.ssl.enabled:true}")
+    private boolean https;
+
+    @Value("${apiml.service.hostname:localhost}")
+    private String hostname;
+
+    @Value("${apiml.service.ipAddress:127.0.0.1}")
+    private String ipAddress;
+
+    @Value("${apiml.service.port:10010}")
+    private int port;
+
+    private InstanceInfo getInstanceInfo(String serviceId) {
+        var leaseInfo = LeaseInfo.Builder.newBuilder()
+            .setDurationInSecs(Integer.MAX_VALUE)
+            .setRegistrationTimestamp(System.currentTimeMillis())
+            .setRenewalTimestamp(System.currentTimeMillis())
+            .setRenewalIntervalInSecs(Integer.MAX_VALUE)
+            .setServiceUpTimestamp(System.currentTimeMillis())
+            .build();
+
+        var scheme = https ? "https" : "http";
+
+        Map<String, String> metadata = new HashMap<>();
+        switch (serviceId) {
+            case "gateway":
+                metadata = eurekaInstanceGw.getMetadataMap();
+                metadata.put("management.port", "10010");
+                break;
+            default:
+        }
+
+        return InstanceInfo.Builder.newBuilder()
+            .setInstanceId(String.format("%s:%s:%d", hostname, serviceId, port))
+            .setAppName(serviceId)
+            .setHostName(hostname)
+            .setHomePageUrl(null, String.format("%s://%s:%d", scheme, hostname, port)) //TODO Will it break something else other than catalog?
```

Comment:
I guess it's solved?

Reply:
I need to improve this, it's a quick fix only, thanks for spotting it